### PR TITLE
Fix poll condition when enable_collection() is not called

### PR DESCRIPTION
**Pull request by qinsoon**
This PR fixes https://github.com/mmtk/mmtk-core/issues/214. We no longer allow allocating without checking budget if the user does not call `enable_collection()`.

### CODE REVIEW

**Review by javadamiri — APPROVED** [review]
LGTM

File: src/policy/space.rs [hunk 1]
```diff
@@ -243,8 +243,13 @@ pub trait Space<VM: VMBinding>: 'static + SFT + Sync + Downcast {
 
         trace!("Polling ..");
 
-        if allow_poll && VM::VMActivePlan::global().poll(false, self.as_space()) {
```
Comment (caizixian) [inline src/policy/space.rs:246]: This is now dead code. Should be removed
```Rust
let allow_poll = unsafe { VM::VMActivePlan::is_mutator(tls) }
            && VM::VMActivePlan::global().is_initialized();
```

**Review by caizixian — CHANGES_REQUESTED** [review]

File: src/policy/space.rs (hunk 1 above)
Comment (qinsoon) [reply src/policy/space.rs:246]: It is not actually dead code. It is still used in the `else` block.

Comment (qinsoon) [reply src/policy/space.rs:246]: I will refactor this part to make it cleaner.

Comment (caizixian) [reply src/policy/space.rs:246]: Ah, sorry, I missed that.

Comment (qinsoon) [reply src/policy/space.rs:246]: Refactored the code a bit, and added some comments to make it clearer.

Comment (qinsoon) [reply src/policy/space.rs:246]: You can take a look and let me know if it is better now (if so, also remove the change request so we can merge this when tests pass).

**Review by caizixian — APPROVED** [review]

File: src/policy/space.rs (hunk 1 above)
Comment (caizixian) [reply src/policy/space.rs:246]: Thanks Yi. It's much clearer now.

**Pull request merged**